sources/updater/update_metadata/src/lib.rs

```diff
@@ -419,35 +452,167 @@ pub fn load_manifest<T: tough::Transport>(repository: &tough::Repository<T>) ->
+        // Each seed will be mapped to a single millisecond within this wave,
+        // e.g. seed 1 -> update is ready 1 millisecond past start of wave
+        // seed 500 -> update is ready 500 millisecond past start of wave, etc
+        waves.insert(0, time);
+        waves.insert(MAX_SEED, time + Duration::milliseconds(MAX_SEED as i64));
+        let mut update = Update {
+            variant: String::from("bottlerocket"),
+            arch: String::from("test"),
+            version: Version::parse("1.1.1").unwrap(),
+            max_version: Version::parse("1.1.1").unwrap(),
+            waves,
+            images: Images {
+                boot: String::from("boot"),
+                root: String::from("root"),
+                hash: String::from("hash"),
+            },
+        };
+
+        for seed in (100..500).step_by(100) {
+            assert!(
+                !update.update_ready(seed, time),
+                "seed: {}, time: {}, wave start time: {}, wave start seed: {}, {} milliseconds hasn't passed yet",
+                seed,
+                time,
+                time,
+                0,
+                seed
+            );
+            assert!(
+                update.update_ready(seed, time + Duration::milliseconds(seed as i64)),
+                "seed: {}, time: {}, wave start time: {}, wave start seed: {}, update should be ready",
+                seed,
+                time + Duration::milliseconds(100),
+                time,
+                0,
+            );
+        }
+
+        // Test multiple waves
+        update.waves.clear();
+        let mut seed = 150;
+        assert!(
+            update.update_ready(seed, time),
+            "No waves specified but no update"
+        );
 
-#[test]
-fn test_migrations_backward() {
-    // The same manifest as `test_migrations_forward` but this time we will migrate backward.
-    let path = "./tests/data/migrations.json";
-    let manifest: Manifest = serde_json::from_reader(File::open(path).unwrap()).unwrap();
-    let from = Version::parse("1.5.0").unwrap();
-    let to = Version::parse("1.0.0").unwrap();
-    let targets = find_migrations(&from, &to, &manifest).unwrap();
-
-    assert!(targets.len() == 3);
-    let mut i = targets.iter();
-    assert!(i.next().unwrap() == "migration_1.5.0_shortcut");
-    assert!(i.next().unwrap() == "migration_1.1.0_b");
-    assert!(i.next().unwrap() == "migration_1.1.0_a");
+        // First wave starts immediately
+        update.waves.insert(0, time);
+        // First wave ends 200 milliseconds after it starts and has seeds 0 - 50
+        update.waves.insert(50, time + Duration::milliseconds(200));
+        // Second wave ends 1024 milliseconds after it starts and has seeds 50 - 100
+        update
+            .waves
+            .insert(100, time + Duration::milliseconds(1024));
+        // Third wave ends 4096 milliseconds after it starts and has seeds 100 - 1024
```

Comment:
"it" here refers to start of the 0th wave (apex/creation of update)... I should make that more clear ugh